iPetCare/Application/Services/ExaminationService.cs

```diff
@@ -0,0 +1,220 @@
+            }
+
+            var examinations = await Context.Examinations.ToListAsync();
+            var filteredExaminations = examinations.Where(ex => ex.PetId.ToString().Equals(petId)).ToList();
+
+            var dto = new ExaminationsGetAllExaminationsDtoResponse()
+            {
+                Examinations = Mapper.Map<List<ExaminationsDetailGetAllDtoResponse>>(filteredExaminations)
+            };
+
+            return new ServiceResponse<ExaminationsGetAllExaminationsDtoResponse>(HttpStatusCode.OK, dto);
+        }
+
+        public async Task<ServiceResponse<ExaminationsUpdateExaminationDtoResponse>> UpdateExaminationAsync(string petId, string examinationId, ExaminationsUpdateExaminationDtoRequest dto)
+        {
+            if (CurrentlyLoggedUser == null)
+                return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.Unauthorized);
+
+            var pet = Context.Pets.Find(Guid.Parse(petId));
+            if (pet == null)
+                return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.NotFound);
+
+            if (!checkIfCanEditExamination(pet))
+            {
+                return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.Forbidden);
+            }
+
+            var examination = Context.Examinations.Find(Guid.Parse(examinationId));
+            var examinationType = Context.ExaminationTypes.Find(dto.ExaminationTypeId);
+            Note note = null;
+            if (dto.NoteId != null)
+                note = Context.Notes.Find(dto.NoteId);
+
+            if (examination == null)
+                return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.NotFound);
+            if (examinationType == null)
+                return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.NotFound);
+            if (note == null)
+                examination.NoteId = null;
+            else
+                examination.NoteId = Guid.Parse(dto.NoteId);
+
+            examination.Date = dto.Date;
+            examination.ExaminationTypeId = dto.ExaminationTypeId;         
+            examination.PetId = Guid.Parse(dto.PetId);
+
+            int result = await Context.SaveChangesAsync();
+            if (result > 0)
+            {
+                var responseDto = Mapper.Map<ExaminationsUpdateExaminationDtoResponse>(examination);
+                return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.OK, responseDto);
+            }
+
+            return new ServiceResponse<ExaminationsUpdateExaminationDtoResponse>(HttpStatusCode.BadRequest);
+        }
+
+        private Boolean checkIfCanEditExamination( Pet pet)
+        {
+            if (CurrentlyLoggedUser.Role != Role.Administrator)
+            {
+                var owners = pet.OwnerPets.Where(ownerpet => ownerpet.OwnerId.ToString().Equals(CurrentlyLoggedUser.Id));
```

Comment:
Ta metoda nie zadziała poprawnie. OwnerId to nie to samo co UserId. Analogicznie z Vetid w linijce poniżej